Control/Concurrent/STM/TSem.hs

```diff
@@ -72,6 +77,29 @@ waitTSem (TSem t) = do
   writeTVar t $! (i-1)
 
 
+-- | Multi-wait on 'TSem'.
+--
+-- Acquire multiple units at once from semaphore; blocks (via 'retry')
+-- when the number of available units is less than the requested
+-- amount.
+--
+-- > waitTSem == waitTSemN 1
+--
+-- __NOTE__: This operation is subtly different from repeatedly
+-- invoking 'waitTSem'. Note that 'waitTSem 0' can block.
+--
+-- @since TBD
+waitTSemN :: Word.Word -> TSem -> STM ()
+waitTSemN 0 (TSem t) = do
```

Comment:
I'm actually considering turning this into a no-op, as this allows to observe an otherwise internal implementation details (i.e. `waitTSemN 0` can only ever block if `newTSem` constructed a `TSem` with a negative counter value; that's currently the only way the internal counter can ever go negative).

Reply:
This seems reasonable to me.